src/fondant/core/manifest.py

```diff
@@ -220,39 +195,61 @@ def cache_key(self) -> str:
         return self.metadata["cache_key"]
 
     @property
-    def index(self) -> Index:
-        return Index(self._specification["index"], base_path=self.base_path)
-
-    @property
-    def subsets(self) -> t.Mapping[str, Subset]:
-        """The subsets of the manifest as an immutable mapping."""
+    def fields(self) -> t.Mapping[str, Field]:
+        """The fields of the manifest as an immutable mapping."""
         return types.MappingProxyType(
             {
-                name: Subset(subset, base_path=self.base_path)
-                for name, subset in self._specification["subsets"].items()
+                name: Field(
+                    name=name,
+                    type=Type(field["type"]),
+                    location=field["location"],
+                )
+                for name, field in self._specification["fields"].items()
             },
         )
 
-    def add_subset(
-        self,
-        name: str,
-        fields: t.Iterable[t.Union[Field, t.Tuple[str, Type]]],
-    ) -> None:
-        if name in self._specification["subsets"]:
-            msg = f"A subset with name {name} already exists"
+    def add_or_update_field(self, field: Field, overwrite: bool = False):
+        """Add or update field to manifest."""
+        if field.name == "index":
+            self._add_or_update_index(field, overwrite=True)
+        elif overwrite is False and field.name in self._specification["fields"]:
+            msg = (
+                f"A field with name {field.name} already exists. Set overwrite to true, "
+                f"if you want to update the field."
+            )
+            raise ValueError(msg)
+        else:
+            self._specification["fields"][field.name] = {
+                "location": f"/{self.component_id}",
+                "type": field.type.name,
```

Comment:
See my comment below. We should store them as json mainly to match the format of the component spec 

```suggestion
                "type": field.type.to_json(),
```

Reply:
Thanks! Wasn't aware of this, but it makes totally sense!

Reply:
This broke some of the tests though.

```python
>>> type.to_json()

{
  "type": type
}
```

So now we get:
```python
{
  "location": ...,
  "type": {
    "type": type
  }
}
```

I think we can solve it like this:

```python
self._specification["fields"][field.name] = {
  "location": f"/{self.component_id}",
  **field.type.name,
}
```